### PR TITLE
JAMES-2895 Remove link to the wiki

### PR DESCRIPTION
**Pull request by chibenwa**
Articles there age from 2007

### CODE REVIEW

File: src/site/site.xml [hunk 1]
```diff
@@ -84,7 +84,6 @@
                 <item name="Mailing Lists" href="/mail.html" />
                 <item name="Contributing" href="/contribute.html" />
                 <item name="Guidelines" href="/guidelines.html" />
-                <item name="Wiki" href="http://wiki.apache.org/james" />
```
Comment (mbaechler) [inline src/site/site.xml]: shoudn't we delete this wiki, too?

Comment (chibenwa) [reply src/site/site.xml]: I will create the INFRA ticket once it is deployed & merged.

**Comment by chibenwa** [comment]
Merged

**Comment by chibenwa** [comment]
See https://issues.apache.org/jira/browse/INFRA-19134 for removal of wiki

**Pull request closed without merging**